# The Weather in Philadelphia

By analyzing the weather in philadelphia dataset, I wanted to look into any patterns/relationships were particularly notable. I wanted to see if there were any associations to myself as well. Overall, I just wanted to observe some collective patterns, and find some associations. 

![](images/philadelphia.jpg){width=500}

Reading the data: 

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [4]:
weather_df = pd.read_csv('data/philadelphia_weather_2005_to_2025.csv')
weather_df = weather_df.assign(date=pd.to_datetime(weather_df['date']))

#### Inspecting the data

In [5]:
weather_df.columns

Index(['day', 'year', 'high', 'low', 'rain', 'snow', 'date', 'day_num',
       'month_num', 'dow', 'dow_name', 'month'],
      dtype='str')

In [6]:
weather_df.head()

,day,year,high,low,rain,snow,date,day_num,month_num,dow,dow_name,month
0,January 1,2005,64,39,0.00,0.0,2005-01-01,1,1,5,Saturday,January
1,January 2,2005,47,33,0.01,0.0,2005-01-02,2,1,6,Sunday,January
2,January 3,2005,57,40,0.06,0.0,2005-01-03,3,1,0,Monday,January
3,January 4,2005,55,44,0.00,0.0,2005-01-04,4,1,1,Tuesday,January
4,January 5,2005,47,36,0.62,0.0,2005-01-05,5,1,2,Wednesday,January


In [7]:
weather_df['year'].value_counts()

year
2008    366
2020    366
2024    366
2016    366
2012    366
2009    365
2007    365
2006    365
2005    365
2013    365
2014    365
2011    365
2010    365
2017    365
2015    365
2019    365
2018    365
2021    365
2022    365
2023    365
2025    365
Name: count, dtype: int64

## 1) What was the weather like in Philadelphia the day of my birthday (01-30) for each year since 2006-2025?
  ## * a. What was the average fahrenheit for the day?
  ## * b. Did it rain?
  ## * c. Did it snow?
  ## * d. What day of the week was it? 

**All information for each of my birthdays (most important include the temperature of high, low, rain, and snow). High represents the highest temperature of the day. Low represents the lowest temperature. Rain and Snow include decimals representing the amount of snow or rain, that can be converted into a percentage.**

In [8]:
weather_df.query('day =="January 30"')

,day,year,high,low,rain,snow,date,day_num,month_num,dow,dow_name,month
29,January 30,2005,36,26,0.15,1.3,2005-01-30,30,1,6,Sunday,January
394,January 30,2006,62,37,0.00,0.0,2006-01-30,30,1,0,Monday,January
759,January 30,2007,39,20,0.02,0.2,2007-01-30,30,1,1,Tuesday,January
1124,January 30,2008,49,29,0.05,0.0,2008-01-30,30,1,2,Wednesday,January
1490,January 30,2009,36,22,0.00,0.0,2009-01-30,30,1,4,Friday,January
1855,January 30,2010,21,16,0.10,2.1,2010-01-30,30,1,5,Saturday,January
2220,January 30,2011,39,18,0.00,0.0,2011-01-30,30,1,6,Sunday,January
2585,January 30,2012,42,33,0.00,0.0,2012-01-30,30,1,0,Monday,January
2951,January 30,2013,68,38,0.32,0.0,2013-01-30,30,1,2,Wednesday,January
3316,January 30,2014,28,10,0.00,0.0,2014-01-30,30,1,3,Thursday,January


![](images/temp.png){width=400}

## A) What was the average fahrenheit for the day? 

In [9]:
birthday = weather_df.query('day == "January 30"') 
temp_avg = (birthday['high'] + birthday['low'])/2
weather_df = weather_df.assign(temp_avg=(weather_df['high']+weather_df['low'])/2)

In [10]:
weather_df.query('day =="January 30"')

,day,year,high,low,rain,snow,date,day_num,month_num,dow,dow_name,month,temp_avg
29,January 30,2005,36,26,0.15,1.3,2005-01-30,30,1,6,Sunday,January,31.0
394,January 30,2006,62,37,0.00,0.0,2006-01-30,30,1,0,Monday,January,49.5
759,January 30,2007,39,20,0.02,0.2,2007-01-30,30,1,1,Tuesday,January,29.5
1124,January 30,2008,49,29,0.05,0.0,2008-01-30,30,1,2,Wednesday,January,39.0
1490,January 30,2009,36,22,0.00,0.0,2009-01-30,30,1,4,Friday,January,29.0
1855,January 30,2010,21,16,0.10,2.1,2010-01-30,30,1,5,Saturday,January,18.5
2220,January 30,2011,39,18,0.00,0.0,2011-01-30,30,1,6,Sunday,January,28.5
2585,January 30,2012,42,33,0.00,0.0,2012-01-30,30,1,0,Monday,January,37.5
2951,January 30,2013,68,38,0.32,0.0,2013-01-30,30,1,2,Wednesday,January,53.0
3316,January 30,2014,28,10,0.00,0.0,2014-01-30,30,1,3,Thursday,January,19.0


In [11]:
birthday_mean_temp = temp_avg.mean()

print("Average birthday temperature:", round(birthday_mean_temp, 1), "degrees")

Average birthday temperature: 32.3 degrees


#### Since I was born in 2006, combining the weather data of Philadelphia for the past twenty years, the average fahrenheit on my birthday in Philadelphia is 32.3 degrees. 

Looking at the weather on my birthday from 2006 to 2025 showed that January 30 in Philadelphia is usually cold, but not always extremely cold. The average temperature across all years was about 32.3 degrees in Fahrenheit. While some birthdays were especially cold, others were surprisingly ok. Winter weather on my birthday was varied more than I expected.

![](images/rain.jpg){width=500}

## B) Did it rain? 

In [13]:
birthday = weather_df.query('day == "January 30"') 
is_rain = birthday['rain']>0

In [14]:
rain_counts = is_rain.value_counts()

rain_yes = rain_counts[True]
rain_no = rain_counts[False]

In [17]:
print(f"Out of 21 birthdays analyzed, it rained on {rain_yes} birthdays and did not rain on {rain_no} birthdays.")

Out of 21 birthdays analyzed, it rained on 8 birthdays and did not rain on 13 birthdays.


**Out of 21 birthdays, it rained on 8 birthdays and did not rain on 13 birthdays.**

Most of my birthdays in Philadelphia did not have rain. Around 62 percent out of all my birthdays recorded rainfall on January 30. Even through the winter, the data showed that rainy birthdays were actually less common for me.

![](images/snow.jpeg){width=500}

## C) Did it snow? 

In [19]:
birthday = weather_df.query('day == "January 30"') 
is_snow = birthday['snow']>0

In [20]:
snow_counts = is_snow.value_counts()

snow_yes = snow_counts[True]
snow_no = snow_counts[False]

In [21]:
print(f"Out of 21 birthdays analyzed, it snowed on {snow_yes} birthdays and did not snow on {snow_no} birthdays.")

Out of 21 birthdays analyzed, it snowed on 7 birthdays and did not snow on 14 birthdays.


**Out of 21 birthdays analyzed, it snowed on 7 birthdays and did not snow on 14 birthdays.**

Snowfall was also less common than I originally thought. Most of my birthdays did not have a lot of snowfall, even though January is in the winter. This was a suprising finding because January 30 is directly in the winter.

## D) What day of the week was it? 

In [22]:
weather_df.query('day == "January 30"') ['dow_name'].value_counts()

dow_name
Monday       4
Sunday       3
Tuesday      3
Wednesday    3
Saturday     3
Thursday     3
Friday       2
Name: count, dtype: int64

#### Since I was born in 2006, the largest amount of my birthdays have fallen on a Monday. 

Mondays appeared more than any other weekday for my birthdays. This was interesting because weekday birthdays often feel less celebratory (and more like going to school) compared to birthdays that fall on Fridays or weekends.

![](images/conclusion.jpg){width=500}

## Conclusion 

The weather in Philadelphia on the days of my birthday for the past twenty years was something I was curious about, especially to see if there was a pattern that I can relate back to myself. I found that majority of my birthdays fell on a Monday, it did snow nor rain for most of the years, and the average temperature was 32 degrees on my birthdays put together. So overall, most of my birthdays throughout the years here in Philadelphia were **relatively chilly**, but not always **snowy or stormy**. I find that interesting, because I spent my recent birthday in 2026 freezing, because of how cold it was from the piled up snow from the previous snow storm and high winds. I thought this was the norm weather here in Philadelphia on Jan 30, but referencing past years' data it shows otherwise. This was an interesting finding. Comparing my memories to the actual data helped me realized that it was more varied than I had expected. 